centos-pypa-rust-x86_64/Dockerfile

```diff
@@ -0,0 +1,26 @@
+FROM quay.io/pypa/manylinux2014_x86_64
+
+WORKDIR /usr/src/app
+
+# Download rustup and checksum
+RUN curl https://static.rust-lang.org/rustup/dist/x86_64-unknown-linux-gnu/rustup-init.sha256 | awk '{print $1 " rustup-init"}' > checksum.txt && \
+    curl -O https://static.rust-lang.org/rustup/dist/x86_64-unknown-linux-gnu/rustup-init
```

Comment:
It would be cool if instead of duplicating the aarch64 dockerfile, maybe this URL could be templated with a variable using docker build args. Maybe not required, but would be neater to avoid common issues with duplication of some Dockerfiles being better supported/updated than another.

Reply:
The existing `arch` matrix variable from the workflow already seems perfect for this

Reply:
Eh this might not always be preferable actually. If we need to add some dependencies to certain architectures ever in the future, it might be nice to keep them as separate dockerfiles. Resolving this one